### PR TITLE
feat: sigmoid activation function

### PR DESCRIPTION
**Pull request by Pochetes**
## Overview
- implemented sigmoid activation function
- instantiated one in `Value` form to make calculation

check if I'm missing anything

### CODE REVIEW

File: src/activation/sigmoid.rs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+use super::Activation;
+use crate::value::Value;
+
+impl Activation {
+    pub fn sigmoid(&self, value: Value) -> Value {
+        let one = Value::new(1.0);
```
Comment (LazaroHurtado) [inline src/activation/sigmoid.rs:6]: There should already be an existing public method that does what you are doing here, so perhaps it's better to replace this logic with `Value::one()`. It's indirect but the overhead is small and dependent.

File: src/activation/activation.rs [hunk 2]
```diff
@@ -8,11 +8,13 @@ pub enum Activation {
     ReLu,
     Tanh,
     Softmax,
+    Sigmoid,
 }
 
 impl Activation {
     pub fn activate<D: Dimension>(&self, inputs: &Tensor<D>) -> Tensor<D> {
         match self {
+            Self::Sigmoid => inputs.mapv(|input| self.sigmoid(input)),
```
Comment (LazaroHurtado) [inline src/activation/activation.rs:17]: Very nitpicky, but can you place this line below the `Softmax` case to follow the same order as the Enum?

**Review by LazaroHurtado — CHANGES_REQUESTED** [review]
I left some small comments and CI tests point out other concerns. Also, this feature should include a test file that compares the `activate` output to what Pytorch produces for the same given input, you can use ReLU test file as a reference!

File: src/activation/sigmoid.rs (hunk 1 above)
Comment (Pochetes) [reply src/activation/sigmoid.rs:6]: ah okay, was trying to find something like this

Comment (Pochetes) [reply src/activation/sigmoid.rs:6]: changed it to a one liner

**Comment by Pochetes** [comment]
Made changes. Tests should be passing and it makes sense that an input of 0 results in 0.5 which is right in the middle of the S curve.

**Review by LazaroHurtado — APPROVED** [review]
LGTM, thanks for adding a test file! The CI tests require approval for first-time collaborators but since I don't see the downside of always running then you are totally right, will be changing it!
